test/buffered_logger/log_device_proxy_slow_test.rb

```diff
@@ -0,0 +1,28 @@
+require "test_helper"
+
+describe "BufferedLogger::LogDeviceProxy slow tests" do
+  before do
+    @logdev = mock()
+    @proxy = BufferedLogger::LogDeviceProxy.new(@logdev)
+  end
+
+  describe "flush" do
+    it "should handle multiple thread simutaneously calling it" do
```

Comment:
This is a brute force test to recreate the issue I was seeing in production. If you remove the fixes in the PR you'll usually get failures and exceptions when running this test.